### PR TITLE
Quality bug fixes

### PR DESCRIPTION
**Pull request by julien-louis**
fix all from https://github.com/fli-iam/shanoir-ng/issues/1617

contains code for "auto check at import" but the front part is commented and should be invisible

### CODE REVIEW

**Comment by jcomedouteau** [comment]
I'm testing it. Did you merge develop @julien-louis  ?

**Comment by jcomedouteau** [comment]
org.springframework.web.client.RestClientException: Can not get dicom attributes for dataset 73; nested exception is org.springframework.web.client.HttpClientErrorException$NotFound: 404 Not Found: "{"errorMessage":"No matches found."}"

image

**Comment by jcomedouteau** [comment]
I have this problem when trying to
datasets                  | java.lang.NullPointerException: null
datasets                  |     at org.shanoir.ng.dataset.security.DatasetSecurityService.hasRightOnSubjectStudies(DatasetSecurityService.java:164)

subjectStudyService.update(result.getUpdatedSubjectStudies());
This may be null ? (which is weird)

**Comment by jcomedouteau** [comment]
As admin, when I launch a quality control, everything is empty (I have two subjects with multiples datasets)
image

**Comment by jcomedouteau** [comment]
Weird display (+ new dataset name ?)
image

**Comment by jcomedouteau** [comment]
When creating a new rule, I can select 'dataset acquisition' but still select a dataset field -> which leads to a bad state where the field is null:
image

**Comment by julien-louis** [comment]
> I have this problem when trying to apply quality card datasets | java.lang.NullPointerException: null datasets | at org.shanoir.ng.dataset.security.DatasetSecurityService.hasRightOnSubjectStudies(DatasetSecurityService.java:164)
> 
> subjectStudyService.update(result.getUpdatedSubjectStudies()); This may be null ? (which is weird)

Are you sure you have rebuild dataset because it's the same error that I fixed in [102f575](https://github.com/fli-iam/shanoir-ng/pull/1654/commits/102f57507dc7e353afeed7cfdec137d6c651bac0)

**Comment by michaelkain** [comment]
Dear Julien (@julien-louis), I hope you are doing fine. This PR is mandatory for the next test session and the go-live. Thank you and with kind regards, Michael

**Comment by jcomedouteau** [comment]
Is it reviewable @julien-louis

**Comment by julien-louis** [comment]
@jcomedouteau now yes

**Comment by jcomedouteau** [comment]
I can't reproduce my first error. I'll check the code rapidly.
I think we can note somewhere to add a "not" to inverse a rule.

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/controler/QualityCardApiController.java [hunk 1]
```diff
@@ -157,8 +157,21 @@ public ResponseEntity<QualityCardResult> applyQualityCardOnStudy(
...
+	
+	@Override
+    public ResponseEntity<QualityCardResult> testQualityCardOnStudy(
+            @ApiParam(value = "id of the quality card", required = true) @PathVariable("qualityCardId") Long qualityCardId) throws RestServiceException, MicroServiceCommunicationException {
+        
+        final QualityCard qualityCard = qualityCardService.findById(qualityCardId);
+        if (qualityCard == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        LOG.info("test quality card: name:" + qualityCard.getName() + ", studyId: " + qualityCard.getStudyId());
```
[the remark this answers is not in the document]
Comment (julien-louis) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/controler/QualityCardApiController.java:172]: Ok, will do in the next quality developments

**Comment by michaelkain** [comment]
Dear @julien-louis, dear @jcomedouteau, thank you for your work on that PR! Could you please deploy it next week, once finished? Thank you and with kind regards, Michael

**Comment by julien-louis** [comment]
Ok so the last bug is because the acquisitions were CT and that non MR Acquisitions are filtered out of the quality control. Let's discuss it.

**Pull request merged**